server/router/services/cluster-service.js

```diff
@@ -19,39 +19,24 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 // THE SOFTWARE.
 
-const { CLUSTER_CACHE_TTL } = require('../constants');
-
-let cache = null;
-let cacheExpiryDateTime = null;
-
-const clearCache = () => {
-  cache = null;
-  cacheExpiryDateTime = null;
-};
-
-const setCache = data => {
-  cache = data;
-  cacheExpiryDateTime = Date.now() + CLUSTER_CACHE_TTL;
-};
-
-const getCluster = async ctx => {
-  if (cacheExpiryDateTime && Date.now() < cacheExpiryDateTime) {
-    return cache;
+class ClusterService {
+  constructor(cacheManager) {
+    this.cacheManager = cacheManager;
   }
 
-  const cluster = await ctx.cadence.describeCluster();
+  fetch(ctx) {
+    return async () => {
```

Comment:
feels sorta odd to have a fetch-func-factory called "fetch", rather than actually fetching + making it lazy where necessary with `async () => fetch(ctx)`

might just be the name tho.

Reply:
Perhaps rename to `fetchClusterHandler`? Basically it's a curried callback which is called by cache manager when the cache has expired / not set.

Reply:
I think it's more that the name doesn't sound like it returns a curried func that can be used to trigger a fetch in the future.  it sounds like it fetches something _now_.

Reply:
Code updated with change: